File: Sorting Algorithms/TreeSort.java
// File Name: TreeSort.java

class TreeSort {

    class Node {
        int data;
        Node left, right;

        Node(int value) {
            data = value;
            left = right = null;
        }
    }

    Node root;

    // Insert into BST
    Node insert(Node root, int data) {
        if (root == null)
            return new Node(data);

        if (data < root.data)
            root.left = insert(root.left, data);
        else
            root.right = insert(root.right, data);

        return root;
    }

    // Inorder Traversal
    void inorder(Node root) {
        if (root != null) {
            inorder(root.left);
            System.out.print(root.data + " ");
            inorder(root.right);
        }
    }

    public static void main(String[] args) {
        int arr[] = {64, 34, 25, 12, 22, 11, 90};

        TreeSort tree = new TreeSort();

        System.out.println("Original Array:");
        for (int value : arr) {
            System.out.print(value + " ");
            tree.root = tree.insert(tree.root, value);
        }

        System.out.println("\nSorted Array:");
        tree.inorder(tree.root);
    }
}

File: Sorting Algorithms/TreeSort.c
/* File Name: TreeSort.c */

#include <stdio.h>
#include <stdlib.h>

// Define BST Node
struct Node {
    int data;
    struct Node* left;
    struct Node* right;
};

// Create new node
struct Node* newNode(int data) {
    struct Node* node = (struct Node*)malloc(sizeof(struct Node));
    node->data = data;
    node->left = node->right = NULL;
    return node;
}

// Insert into BST
struct Node* insert(struct Node* root, int data) {
    if (root == NULL)
        return newNode(data);

    if (data < root->data)
        root->left = insert(root->left, data);
    else
        root->right = insert(root->right, data);

    return root;
}

// Inorder Traversal (Sorted Order)
void inorder(struct Node* root) {
    if (root != NULL) {
        inorder(root->left);
        printf("%d ", root->data);
        inorder(root->right);
    }
}

int main() {
    int arr[] = {64, 34, 25, 12, 22, 11, 90};
    int n = sizeof(arr) / sizeof(arr[0]);

    struct Node* root = NULL;

    printf("Original Array:\n");
    for (int i = 0; i < n; i++) {
        printf("%d ", arr[i]);
        root = insert(root, arr[i]);
    }

    printf("\nSorted Array:\n");
    inorder(root);

    return 0;
}
